### PR TITLE
Pull request saiga

### PR DESCRIPTION
**Pull request by Paul-Hi**
Pull request without cluster samples.

### CODE REVIEW

File: shader/lighting/light_assignment.glsl [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+ * Licensed under the MIT License.
+ * See LICENSE file for more information.
+ */
+
+##GL_COMPUTE_SHADER
+#version 430 core
+
+#include "camera.glsl"
+
+struct LightClusterData
```
Comment (darglein) [inline shader/lighting/light_assignment.glsl:12]: Same name as in c++

File: shader/lighting/light_assignment.glsl [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+layout (std140, binding = 6) uniform lightInfoBlock
+{
+    int pointLightCount;
+    int spotLightCount;
+    int directionalLightCount;
+
+    int clusterEnabled;
+};
+
+struct cluster
```
Comment (darglein) [inline shader/lighting/light_assignment.glsl:26]: First letter capital

File: shader/lighting/light_assignment.glsl [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    int tileDebug;
+    int splitDebug;
+};
+
+layout (std430, binding = 8) buffer clusterBuffer
+{
+    cluster clusterList[];
+};
+
+struct clusterItem
```
Comment (darglein) [inline shader/lighting/light_assignment.glsl:56]: ^

File: shader/lighting/light_assignment.glsl [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    clusterItem itemList[];
+};
+
+// Shader storage buffers are enabled
+layout (std430, binding = 10) buffer lightClusterData
+{
+    LightClusterData clusterData[];
+};
+
+struct clusterBounds
```
Comment (darglein) [inline shader/lighting/light_assignment.glsl:72]: .

File: shader/lighting/lighting_uber.glsl [hunk 5]
```diff
@@ -31,6 +31,13 @@ void main()
 
 layout(location = 0) out vec4 out_color;
 
+float lin(float d){
```
Comment (darglein) [inline shader/lighting/lighting_uber.glsl:34]: What is this function?

File: shader/lighting/uber_lighting_helpers.glsl [hunk 6]
```diff
@@ -97,9 +91,15 @@ layout (std430, binding = 8) buffer clusterBuffer
 
 struct clusterItem
```
Comment (darglein) [inline shader/lighting/uber_lighting_helpers.glsl:92]: name

File: shader/lighting/uber_lighting_helpers.glsl [hunk 7]
```diff
@@ -97,9 +91,15 @@ layout (std430, binding = 8) buffer clusterBuffer
...
 {
-    int lightIdx;
+    int lightIndex;
 };
 
+int lightIdxFromItem(in int item, in int part)
```
Comment (darglein) [inline shader/lighting/uber_lighting_helpers.glsl:97]: unused?

File: shader/lighting/uber_lighting_helpers.glsl [hunk 8]
```diff
@@ -141,33 +171,33 @@ vec3 calculatePointLights(AssetMaterial material, vec3 position, vec3 normal, fl
...
-    {
-        return vec3(1, 0, 0);
-    }
-    if(clusterIndex < 0)
-    {
-        return vec3(1, 0, 0);
-    }
+    // if(clusterIndex > clusterListCount - 1)
+    // {
+    //     return vec3(0, 1, 0);
```
Comment (darglein) [inline shader/lighting/uber_lighting_helpers.glsl:182]: remove unused code

File: shader/lighting/uber_lighting_helpers.glsl [hunk 9]
```diff
@@ -360,12 +388,14 @@ vec3 calculateDirectionalLights(AssetMaterial material, vec3 position, vec3 norm
...
         float Iemissive = material.data.y;
+        Iemissive = 0;
 
         vec3 color = lightColorDiffuse.rgb *
                     (Idiff * material.color.rgb + Ispec * lightColorSpecular.w * lightColorSpecular.rgb + Iamb * material.color.rgb) +
                     Iemissive * material.color.rgb;
 
         result += color;
     }
+    //result = vec3(0);
```
Comment (darglein) [inline shader/lighting/uber_lighting_helpers.glsl:399]: ?

**Review by darglein — CHANGES_REQUESTED** [review]
Also remove sponza.obj from git.

File: shader/lighting/uber_lighting_helpers.glsl (hunk 7 above)
Comment (Paul-Hi) [reply shader/lighting/uber_lighting_helpers.glsl:97]: Removed. Was only in use when I checked the int32 to two int16 indices.

File: shader/lighting/lighting_uber.glsl (hunk 5 above)
Comment (Paul-Hi) [reply shader/lighting/lighting_uber.glsl:34]: Removed. In use when I did not know about existing helper functions.

**Pull request merged**